### PR TITLE
Added metrics for system CPU usage

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Added metrics for the system CPU usage:
- `arangodb_server_statistics_user_percent`: Percentage of time that the system CPUs have spent in user mode
- `arangodb_server_statistics_system_percent`: Percentage of time that the system CPUs have spent in kernel mode
- `arangodb_server_statistics_idle_percent`: Percentage of time that the system CPUs have been idle
- `arangodb_server_statistics_iowait_percent`: Percentage of time that the system CPUs have been waiting for I/O

These metrics resemble the overall CPU usage metrics in `top`.
They are available on Linux only.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new C++ **Unit tests**

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12586/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12589/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12607/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12614/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12783/

**Comment by mpoeter** [comment]
Next try: http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12792/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12796/

File: lib/ApplicationFeatures/CpuUsageFeature.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGODB_APPLICATION_FEATURES_CPU_USAGE_FEATURE_H
+#define ARANGODB_APPLICATION_FEATURES_CPU_USAGE_FEATURE_H 1
+
+#include "ApplicationFeatures/ApplicationFeature.h"
+#include "Basics/CpuUsageSnapshot.h"
+#include "Basics/Mutex.h"
+
+#include <cstdio>
```
Comment (dothebart) [inline lib/ApplicationFeatures/CpuUsageFeature.h:31]: this include is probably not needed here?

**Review by dothebart — APPROVED** [review]
please check whether it works without <cstdio> or whether it can be moved to the cpp file.

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12805/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12817/

**Comment by mpoeter** [comment]
tests blue

**Pull request merged**